### PR TITLE
Remove need to explicitly map :myapp and MyApp.Repo in Migrations documentation

### PR DESCRIPTION
**Pull request by aforward**
### Summary of changes

Great documentation, thanks!

I generalized the Ecto DB migration page so that the code can leverage existing configurations.  In particular, the `:ecto_repos` defined in

```
config :myapp, ecto_repos: [MyApp.Repo]
```

And extracting the application from the `__MODULE__`

The new code is a little more abstract, which might make it a bit harder for users to understand, but I think it makes good use of existing information to avoid duplication.

I only tested the "migrate" code (as I am currently NOT integrating seeds into my release process), I am not sure if you had a nice sandbox to easily test, or if you want to push back onto me to test.

Here's the output from my `migrate`  based on the code I updated.

```
iex(1)> Nameui.ReleaseTasks.migrate
Running migrations for namedb
[info] Already up
:ok
```

### CODE REVIEW

File: docs/Running Migrations.md [hunk 1]
```diff
@@ -26,30 +26,32 @@ defmodule MyApp.ReleaseTasks do
...
-    :myapp
-  ]
 
-  @repos [
-    MyApp.Repo
-  ]
+  def myapp do
+    __MODULE__
+    |> :application.get_application
+    |> (fn {:ok, app} -> app end).()
```
Comment (michalmuskala) [inline docs/Running Migrations.md]: Wouldn't plain old
```
{:ok, app} = :application.get_application(__MODULE__)
app
```
be actually more reasable?

Comment (aforward) [reply docs/Running Migrations.md]: I like to see all things are transformations, and maybe things too far (along the same lines as Dave Thomas, https://vimeo.com/216107561)

That said, I can change it to the simpler code (originally I didn't have `myapp` so the piping did more, but now that it's only the one call... just teasing it out would be better)

Comment (aforward) [reply docs/Running Migrations.md]: Change updated.

File: docs/Running Migrations.md [hunk 2]
```diff
@@ -26,30 +26,31 @@ defmodule MyApp.ReleaseTasks do
...
-  
-  @myapps [
-    :myapp
-  ]
 
-  @repos [
-    MyApp.Repo
-  ]
+  def myapp do
+    {:ok, app} = :application.get_application(__MODULE__)
```
Comment (bitwalker) [inline docs/Running Migrations.md]: Can you make this `Application.get_application(__MODULE__)` for consistency?

File: docs/Running Migrations.md [hunk 3]
```diff
@@ -26,30 +26,31 @@ defmodule MyApp.ReleaseTasks do
...
     Enum.each(@start_apps, &Application.ensure_all_started/1)
 
     # Start the Repo(s) for myapp
     IO.puts "Starting repos.."
-    Enum.each(@repos, &(&1.start_link(pool_size: 1)))
+    Enum.each(repos(), &(&1.start_link(pool_size: 1)))
 
     # Run migrations
-    Enum.each(@myapps, &run_migrations_for/1)
+    migrate
```
Comment (bitwalker) [inline docs/Running Migrations.md]: Needs parentheses, e.g. `migrate()`

File: docs/Running Migrations.md [hunk 4]
```diff
@@ -63,11 +64,17 @@ defmodule MyApp.ReleaseTasks do
...
 
+  def migrate, do: Enum.each(repos(), &run_migrations_for/1)
+
   def priv_dir(app), do: "#{:code.priv_dir(app)}"
 
-  defp run_migrations_for(app) do
-    IO.puts "Running migrations for #{app}"
-    Ecto.Migrator.run(MyApp.Repo, migrations_path(app), :up, all: true)
+  defp run_migrations_for(repo) do
+    repo.config
```
Comment (bitwalker) [inline docs/Running Migrations.md]: I'm not a huge fan of forcing things in to pipes, can you modify this to be:

```elixir
app = Keyword.get(repo.config, :otp_app)
IO.puts "Running migrations for #{app}"
Ecto.Migrator.run(repo, migrations_path(app), :up, all: true)
```

**Comment by bitwalker** [comment]
Thanks for this! Just a few comments and I'll get this merged :)

**Comment by bitwalker** [comment]
ping :)

**Comment by aforward** [comment]
Thanks!!! Probably not until this weekend

File: docs/Running Migrations.md (hunk 2 above)
Comment (aforward) [reply docs/Running Migrations.md]: applied

File: docs/Running Migrations.md (hunk 3 above)
Comment (aforward) [reply docs/Running Migrations.md]: applied

File: docs/Running Migrations.md (hunk 4 above)
Comment (aforward) [reply docs/Running Migrations.md]: applied

**Comment by aforward** [comment]
@bitwalker Hey Paul, I cleaned up the code based on your recommendations; I think it should be good to merge now; let me know if there is anything you would like to see.

**Comment by bitwalker** [comment]
Thanks for this!

**Pull request merged**